### PR TITLE
Some tools to dump other trees.

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Comment by Hixie** [comment]
(The semantics tree in particular is going to be helpful in driving down a11y bugs.)

**Comment by Hixie** [comment]
cc @tvolkert

File: packages/flutter/test/foundation/service_extensions_test.dart [hunk 1]
```diff
@@ -142,16 +142,52 @@ void main() {
...
+    console.clear();
+  });
+
+  test('Service extensions - debugDumpSemanticsTree', () async {
+    Map<String, String> result;
+
+    await binding.doFrame();
+    result = await binding.testExtension('debugDumpSemanticsTree', <String, String>{});
+    expect(result, <String, String>{});
+    expect(console, <String>['Semantics not collected.']);
```
Comment (tvolkert) [inline packages/flutter/test/foundation/service_extensions_test.dart]: That's unfortunate

Comment (Hixie) [inline packages/flutter/test/foundation/service_extensions_test.dart]: There are separate tests to verify the behavior when semantics are enabled. This just verifies that the callback itself fires.

**Comment by tvolkert** [comment]
LGTM

**Pull request merged**